rules/Php81/NodeFactory/EnumFactory.php

```diff
@@ -25,21 +32,47 @@ public function createFromClass(Class_ $class): Enum_
 
         // constant to cases
         foreach ($class->getConstants() as $classConst) {
-            $enum->stmts[] = $this->createEnumCase($classConst);
+            $enum->stmts[] = $this->createEnumCaseFromConst($classConst);
         }
 
         return $enum;
     }
 
-    private function createEnumCase(ClassConst $classConst): EnumCase
+    public function createFromSpatieClass(Class_ $class): Enum_
+    {
+        $shortClassName = Strings::before($this->nodeNameResolver->getShortName($class), 'Enum');
```

Comment:
On more think, I think `Enum` suffix should be kept, as it will require file name change:

```suggestion
        $shortClassName = $this->nodeNameResolver->getShortName($class);
```